### PR TITLE
add numel API for paddle test=develop

### PR DESCRIPTION
**Pull request by wangchaochaohu**
### PR types

New features 
### PR changes

OPs
### Describe

add numel OP （this api has the same fucntion as size Op which is to calculate the number of element in the tensor）

### CODE REVIEW

**Review by heavengate — DISMISSED** [review]

File: python/paddle/tensor/stat.py [hunk 1]
```diff
@@ -244,3 +245,42 @@ def std(input, axis=None, keepdim=False, unbiased=True, out=None, name=None):
...
+        TypeError: If ``x`` is not a Variable.
+
+
+    Examples:
+        .. code-block:: python
+
+        import paddle
+        
+        paddle.disable_static()
+        numel = paddle.numel(x)
```
Comment (lanxianghit) [inline python/paddle/tensor/stat.py]: 示例代码能运行吗？x没有定义吧？而且也没有给返回值

Comment (wangchaochaohu) [reply python/paddle/tensor/stat.py]: done

File: python/paddle/tensor/stat.py [hunk 2]
```diff
@@ -244,3 +245,43 @@ def std(input, axis=None, keepdim=False, unbiased=True, out=None, name=None):
...
         return tmp
+
+
+def numel(x, name=None):
+    """
+    Returns the number of elements for a tensor, which is a int64 Tensor with shape [1] in static mode
+    or a scalar value in imperative mode
+
+    Args:
+        input (Variable): The input variable.
```
Comment (lanxianghit) [inline python/paddle/tensor/stat.py]: 这里应该写variable还是tensor？让pm确认一下吧

**Review by lanxianghit — DISMISSED** [review]
LGTM

File: python/paddle/tensor/stat.py (hunk 2 above)
Comment (wangchaochaohu) [reply python/paddle/tensor/stat.py]: Done已改成Tensor

**Review by lanxianghit — APPROVED** [review]
LGTM

**Review by TCChenlong — APPROVED** [review]
LGTM

**Review by zhupengyang — APPROVED** [review]

**Pull request merged**